### PR TITLE
Add High-level requestframework

### PR DESCRIPTION
**Pull request by IsaiahDahlberg**
Here is a High-level request framework proof of concept.

I created a second ExampleApp that shows how to use the HttpServices

### CODE REVIEW

**Comment by ShreyaThapa** [comment]
Awesome! Thanks, @IsaiahDahlberg! 🙌 

We've got this on our radar and will be reviewing it in our upcoming sprint.  We'll be sure to keep you posted on any feedback we may have.

Once again, thank you for your valuable contribution! 🙏

**Comment by IsaiahDahlberg** [comment]
@ShreyaThapa, this PR has been updated with the last of the missing HttpServices and is ready for another review!

**Comment by ShreyaThapa** [comment]
Hi @IsaiahDahlberg!

🎉 First of all, a big thanks to you and @miccav01 for your contribution! We really appreciate your effort and the time you've put into this pull request.

Right now, our team is heads down, focusing on initiatives with our new product - [Dwolla Connect](https://www.dwolla.com/dwolla-connect/). As a result, our review process might experience some delay. Rest assured, your pull request is important to us and we will definitely get to it in our upcoming sprints.

Thanks again for your patience and understanding. We're excited to dive into your code soon!

File: Dwolla.Client/HttpServices/DocumentsHttpService.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        }
+
+        public async Task<RestResponse<GetDocumentsResponse>> GetDocumentCollectionAsync(string customerId, CancellationToken cancellation = default)
+        {
+            if (string.IsNullOrWhiteSpace(customerId))
+            {
+                throw new ArgumentException("CustomerId should not be blank.");
+            }
+
+            return await GetAsync<GetDocumentsResponse>(new Uri($"{client.ApiBaseAddress}/customers/{customerId}"), cancellation);
```
Comment (ShreyaThapa) [inline Dwolla.Client/HttpServices/DocumentsHttpService.cs]: Looks like`/documents` is missing from the endpoint for listing documents, so I added a suggestion. 
```suggestion
            return await GetAsync<GetDocumentsResponse>(new Uri($"{client.ApiBaseAddress}/customers/{customerId}/documents"), cancellation);
```

File: ExampleApp.HttpServices/Tasks/Documents/Create.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        public override async Task Run()
+        {
+            Write("Customer ID for whom to upload a document: ");
+            var input = ReadLine();
+
+            using var fileStream = typeof(Create).GetTypeInfo().Assembly
+                .GetManifestResourceStream($"ExampleApp.HttpServices.{FilenameSuccess}");
+
+            var response = await HttpService.Documents.UploadDocumentAsync(
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Documents/Create.cs:21]: It looks like the document isn't successfully uploading to the API. It fails with a `400` HTTP status code error.

File: Dwolla.Client/HttpServices/BeneficialOwnersHttpService.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        }
+
+        public async Task<RestResponse<EmptyResponse>> DeleteBeneficialOwnerAsync(string beneficialOwnerId)
+        {
+            if (string.IsNullOrWhiteSpace(beneficialOwnerId))
+            {
+                throw new ArgumentException("BeneficialOwnerId should not be blank.");
+            }
+
+            return await PostAsync<EmptyResponse>(new Uri($"{client.ApiBaseAddress}/beneficial-owners/{beneficialOwnerId}"), null);
```
Comment (ShreyaThapa) [inline Dwolla.Client/HttpServices/BeneficialOwnersHttpService.cs]: Changing request type to DELETE as per the [docs](https://developers.dwolla.com/docs/balance/api-reference/beneficial-owners/remove).
```suggestion
            return await DeleteAsync<EmptyResponse>(new Uri($"{client.ApiBaseAddress}/beneficial-owners/{beneficialOwnerId}"), null);
```

File: Dwolla.Client/HttpServices/CustomersHttpService.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+		public async Task<RestResponse<IavTokenResponse>> GetCustomerIavTokenAsync(string customerId, CancellationToken cancellation = default)
+		{
+			if (string.IsNullOrWhiteSpace(customerId))
+			{
+				throw new ArgumentException("CustomerId should not be blank.");
+			}
+
+			return await GetAsync<IavTokenResponse>(new Uri($"{client.ApiBaseAddress}/customers/{customerId}/iav-token"), cancellation);
+		}
```
Comment (ShreyaThapa) [inline Dwolla.Client/HttpServices/CustomersHttpService.cs]: Suggestion to remove as IAV token endpoint has now been removed from the API as part of [dwolla.js sunset](https://developers.dwolla.com/changelog#2023-08-31).

```suggestion

```

File: ExampleApp.HttpServices/Tasks/Customers/IavToken.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+        public override async Task Run()
+        {
+            Write("Customer ID for whom to get an IAV token: ");
+            var input = ReadLine();
+
+            var response = await HttpService.Customers.GetCustomerIavTokenAsync(input);
+            WriteLine($"Token created: {response.Content.Token}");
+        }
+    }
+}
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Customers/IavToken.cs]: Suggestion to remove this Task file as IAV token endpoint has now been removed from the API as part of [dwolla.js sunset](https://developers.dwolla.com/changelog#2023-08-31).

File: ExampleApp.HttpServices/Tasks/Labels/Create.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                new CreateLabelRequest
+                {
+                    Amount = new Money { Currency = "USD", Value = amount }
+                }
+            );
+
+            if (response == null) return;
+
+            var ownerResponse = await HttpService.Labels.GetLabelAsync(response.Response.Headers.Location.ToString().Split('/').Last());
+            WriteLine($"Created: {ownerResponse.Content.Id} ");
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Labels/Create.cs]:
```suggestion
            var labelResponse = await HttpService.Labels.GetLabelAsync(response.Response.Headers.Location.ToString().Split('/').Last());
            WriteLine($"Created: {labelResponse.Content.Id} ");
```

File: ExampleApp.HttpServices/Tasks/Labels/Create.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
+using Dwolla.Client.Models;
+using Dwolla.Client.Models.Requests;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Labels
+{
+    [Task("cl", "Create a Label for a Customer")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Labels/Create.cs:9]: Label is created but an unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Labels/CreateLabelLedger.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
+using Dwolla.Client.Models;
+using Dwolla.Client.Models.Requests;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Labels
+{
+    [Task("clle", "Create Label Ledger Entry")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Labels/CreateLabelLedger.cs:9]: Label Ledger Entry is created but an unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Labels/Get.cs [hunk 9]
```diff
@@ -0,0 +1,18 @@
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Labels
+{
+    [Task("gl", "Get Label")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Labels/Get.cs:5]: Unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Labels/GetLabelLedger.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Labels
+{
+    [Task("glle", "Get Label Ledger entry")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Labels/GetLabelLedger.cs:5]: Unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Labels/GetLabelReallocation.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Labels
+{
+    [Task("glr", "Get Label Reallocation")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Labels/GetLabelReallocation.cs:5]: Unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Labels/List.cs [hunk 12]
```diff
@@ -0,0 +1,19 @@
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Labels
+{
+    [Task("ll", "List Labels for a Customer")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Labels/List.cs:5]: Unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Labels/ListLabelLedger.cs [hunk 13]
```diff
@@ -0,0 +1,19 @@
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Labels
+{
+    [Task("llle", "List Label Ledger entries for a Label")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Labels/ListLabelLedger.cs:5]: Unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Labels/Remove.cs [hunk 14]
```diff
@@ -0,0 +1,19 @@
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Labels
+{
+    [Task("rl", "Remove Label")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Labels/Remove.cs:6]: Unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Transfers/Create.cs [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+using Dwolla.Client.Models.Responses;
+using Dwolla.Client.Rest;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Transfers
+{
+    [Task("ct", "Create Transfer")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Transfers/Create.cs:12]: Unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Transfers/Get.cs [hunk 16]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Transfers
+{
+    [Task("gt", "Get Transfer")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Transfers/Get.cs:5]: Unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Masspayments/Get.cs [hunk 17]
```diff
@@ -0,0 +1,18 @@
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Masspayments
+{
+    [Task("gmp", "Get Masspayment")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Masspayments/Get.cs:5]: Unhandled exception is thrown.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

File: ExampleApp.HttpServices/Tasks/Masspayments/Create.cs [hunk 18]
```diff
@@ -0,0 +1,62 @@
+using Dwolla.Client.Models;
+using Dwolla.Client.Models.Requests;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ExampleApp.HttpServices.Tasks.Masspayments
+{
+    [Task("cmp", "Create Masspayment")]
```
Comment (ShreyaThapa) [inline ExampleApp.HttpServices/Tasks/Masspayments/Create.cs:10]: Request is successful but unhandled exception is thrown in console.

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Object reference not set to an instance of an object.)
```

**Comment by ShreyaThapa** [comment]
Hi @IsaiahDahlberg and @miccav01!

🎉 Thank you once again for your valuable contribution to our project! We've thrilled to finally have had a chance to prioritize this PR and have reviewed it. Your effort is greatly appreciated.

While reviewing, we've noticed a few minor issues, which we've highlighted in the comments above along with some suggested quick fixes. Also, it seems the test file isn't passing smoothly. Did you encounter any challenges running it? Perhaps it might need an update to accommodate the new HTTP Services structure.

Additionally, we have made note of some additional schemas and models to add and update. We're in the process of creating tickets on our board to address these, aligning with our high-level methods documentation and the updated [OpenAPI spec](https://github.com/Dwolla/dwolla-openapi).

Regarding the release plan for this change, here's what we were planning: Once approved, we'll merge these updates along with any changes Dwolla makes related to high-level methods into an intermediary branch. This branch will be marked as a Beta pre-release and published to NuGet for developers to integrate into their applications. We'll be actively collecting feedback during this phase to ensure everything runs smoothly. Once we've gathered sufficient usage and feedback, we'll proceed with the main release.

We'd love to hear your thoughts on this plan!

Thank you again for your patience and understanding throughout this process. Your efforts are truly appreciated! 🙌

**Comment by isaiah-eventlink** [comment]
@ShreyaThapa, that plans sounds good.

I resolved all the comments. This PR should be ready for another review. Although, I wasn't able to fully test any Label functionality as its behind a pay wall. But, those tasks should be working fine as it seemed the majority of the issues were all related.

**Comment by ShreyaThapa** [comment]
Thanks for the new commits addressing our comments, @isaiah-eventlink! 🙌 
We'll take a look and make sure to test the labels endpoints as well!

**Comment by natehitze-eventlink** [comment]
@ShreyaThapa Any idea on a timeline for merge or review? We're going to need this soon so we're trying to make a decision on whether to use a private fork or not.

**Comment by ShreyaThapa** [comment]
Hi @natehitze-eventlink! Thank you for bringing this to our attention. We've pulled this into our current sprint, which is scheduled to end this month. We will re-review for merging and releasing this branch as a beta release on NuGet for you to import and use in your app!

**Comment by ShreyaThapa** [comment]
Hi @natehitze-eventlink and @isaiah-eventlink!

Thank you for addressing the comments I made during my last review! I appreciate the effort you put into resolving them.

I am currently running into some issues when trying to pass the tests using `dotnet test`. It’s possible that I might be doing something wrong on my end. Could you provide some guidance on how to properly run the tests? Once I can get the tests to pass, I will be happy to approve the PR and publish it!

Thanks again for your contribution.

**Comment by isaiah-eventlink** [comment]
@ShreyaThapa, the tests were missing a callback. I don't recall how they ended up that way. Anyawho, the tests should now be passing for you

**Comment by ShreyaThapa** [comment]
Awesome! Thanks for the quick update, @isaiah-eventlink! 

I have confirmed that the tests are running and passing for me. I've created a new branch [v7.0.0-alpha](https://github.com/Dwolla/dwolla-v2-csharp/tree/v7.0.0-alpha) for all changes related to high level methods. Would you mind re-opening the PR against that branch? After that, I can go ahead and approve/merge this branch, and work on publishing it to NuGet.

**Comment by isaiah-eventlink** [comment]
@ShreyaThapa, here is the second PR you requested https://github.com/Dwolla/dwolla-v2-csharp/pull/58

**Comment by ShreyaThapa** [comment]
Perfect! Thank you @isaiah-eventlink!

I've merged your branch to the new [v7.0.0-alpha](https://github.com/Dwolla/dwolla-v2-csharp/tree/v7.0.0-alpha) branch. I'll close out this PR. Will let you know when it's published! 🙌

**Pull request closed without merging**